celery/backends/base.py

```diff
@@ -647,11 +647,36 @@ def _forget(self, task_id):
 
     def _store_result(self, task_id, result, state,
                       traceback=None, request=None, **kwargs):
+
+        if state in self.READY_STATES:
+            date_done = datetime.datetime.utcnow()
+        else:
+            date_done = None
+
         meta = {
-            'status': state, 'result': result, 'traceback': traceback,
-            'children': self.current_task_children(request),
+            'status': state,
+            'result': result,
+            'traceback': traceback,
             'task_id': bytes_to_str(task_id),
+            'date_done': date_done,
         }
+
+        if request:
+            request_meta = {
+                # Some unit tests mock the properties, casting to string to
+                # avoid serialization errors
+                'children': self.current_task_children(request),
+                'name': str(getattr(request, 'task', None)),
```

Comment:
Perhaps all this logic here can be transferred as a method of `Context` and return this dictionary?

Reply:
Good idea, but some of the tests don't send Context class as request :(

Reply:
Great, thanks for checking it! :smile: